plugins/hls-class-plugin/src/Ide/Plugin/Class/CodeAction.hs

```diff
@@ -0,0 +1,187 @@
+        methodDiags = filter (\d -> isClassMethodWarning (d ^. J.message)) ghcDiags
+
+        mkActions docPath diag = do
+            ident <- findClassIdentifier docPath range
+            cls <- findClassFromIdentifier docPath ident
+            InstanceBindTypeSigsResult sigs <- handleMaybeM "Unable to GetInstanceBindTypeSigs"
+                $ liftIO
+                $ runAction "classplugin.codeAction.GetInstanceBindTypeSigs" state
+                $ use GetInstanceBindTypeSigs docPath
+            pure $ concatMap mkAction $ minDefToMethodGroups range sigs . classMinimalDef $ cls
+            where
+                range = diag ^. J.range
+
+                mkAction :: [(T.Text, T.Text)] -> [Command |? CodeAction]
+                mkAction methodGroup
+                    = [ mkCodeAction title
+                            $ mkLspCommand plId codeActionCommandId title
+                                (Just $ mkCmdParams methodGroup False)
+                    , mkCodeAction titleWithSig
+                            $ mkLspCommand plId codeActionCommandId titleWithSig
+                                (Just $ mkCmdParams methodGroup True)
+                    ]
+                    where
+                        title = mkTitle $ fst <$> methodGroup
+                        titleWithSig = mkTitleWithSig $ fst <$> methodGroup
+
+                mkTitle methodGroup
+                    = "Add placeholders for "
+                        <> mconcat (intersperse ", " (fmap (\m -> "'" <> m <> "'") methodGroup))
+
+                mkTitleWithSig methodGroup = mkTitle methodGroup <> " with signature(s)"
+
+                mkCmdParams methodGroup withSig =
+                    [toJSON (AddMinimalMethodsParams uri range (List methodGroup) withSig)]
+
+                mkCodeAction title cmd
+                    = InR
+                    $ CodeAction
+                        title
+                        (Just CodeActionQuickFix)
+                        (Just (List []))
+                        Nothing
+                        Nothing
+                        Nothing
+                        (Just cmd)
+                        Nothing
+
+        findClassIdentifier docPath range = do
+            (hieAstResult, pmap) <- handleMaybeM "Unable to GetHieAst"
+                . liftIO
+                . runAction "classplugin.findClassIdentifier.GetHieAst" state
+                $ useWithStale GetHieAst docPath
+            case hieAstResult of
+                HAR {hieAst = hf} ->
+                    pure
+                        $ head . head
+                        $ pointCommand hf (fromJust (fromCurrentRange pmap range) ^. J.start & J.character -~ 1)
+                        ( (Map.keys . Map.filter isClassNodeIdentifier . getNodeIds)
+                            <=< nodeChildren
+                        )
+
+        findClassFromIdentifier docPath (Right name) = do
+            (hscEnv -> hscenv, _) <- handleMaybeM "Unable to GhcSessionDeps"
```

Comment:
This view pattern seems weird. Isn't it just renaming the variable? Can't you just do that with a normal pattern i.e. `(hscenv, _) <- ...`?